Chessington.GameEngine/Pieces/Piece.cs

```diff
@@ -9,16 +9,28 @@ public abstract class Piece
         protected Piece(Player player)
         {
             Player = player;
+            HasMoved = false;
         }
 
         public Player Player { get; private set; }
+        public bool HasMoved { get; private set; }
 
         public abstract IEnumerable<Square> GetAvailableMoves(Board board);
 
         public void MoveTo(Board board, Square newSquare)
         {
             var currentSquare = board.FindPiece(this);
             board.MovePiece(currentSquare, newSquare);
+            HasMoved = true;
+        }
+
+        public void AddIfOnBoard(Square square, List<Square> moves)
```

Comment:
This method doesn't have anything to do with the `Piece` class as far as I can tell? So I don't think it belongs here 🙂

Reply:
This method was used to make sure when I add a move to the move list its a move that isn't off the board. Because it was used/will be used by multiple pieces (queen, bishop, rook etc) I thought it best to put it in the piece class rather than repeat it per individual piece. If I don't keep it here where would you suggest I place it so that all pieces can access/use it?